src/UCTNode.cpp

```diff
@@ -268,28 +236,10 @@ void UCTNode::randomize_first_proportionally() {
         return;
     }
 
+    assert(m_children.size() >= index);
```

Comment:
This test never fails.
`m_children` has always the same size as `accum_vector`. So `index` is always a valid index into `m_children`.

Reply:
I generally use asserts to test my assumptions by adding statements I believe will always to be true if the code is working but which might get broken from bugs, changing code. If you feel strongly I can remove the assert.

Reply:
I like the use of asserts because it document assumed invariants and serves as testing at the same time. Keep that stuff.